site/docs/v1/tech/admin-guide/upgrade.adoc

```diff
@@ -4,44 +4,300 @@ title: Upgrade FusionAuth
+
+=== Migrating Configuration Between Environments
+
+There are a number of options for managing the configuration of your FusionAuth instance over time and promoting changes from one environment to another. These include:
+
+- Script the changes.
+- Terraform.
+- Pulumi.
+- Manual configuration via the UI.
+
+You can read more in link:/docs/v1/tech/admin-guide/configuration-management[the Configuration Management guide].
+
+=== Themes and Customizations
+
+Review custom themes and how they will be impacted by any FusionAuth updates. If you have made any customizations, you may need to update them to work with the new version of FusionAuth. You can use the https://github.com/FusionAuth/fusionauth-theme-helper[Theme Helper] project to perform diffs on the themes to identify any changes that need to be made.
+
+include::docs/v1/tech/installation-guide/_theme-upgrade.adoc[]
+
+== Upgrade Options
+
+=== Easiest Route
+
+The easiest upgrade route is to simply stop all your services, upgrade everything, and then bring it all back online. 
+
+This is a good option for single-instance development or QA systems, or for systems that can tolerate longer periods of downtime.
+
+If minimizing downtime is a goal, then rolling upgrades or a blue-green deployment might be more suitable. 
+
+=== Rolling Upgrade
+
+You can perform a rolling upgrade if you have three FusionAuth nodes running behind a load balancer. The nodes are upgraded one at a time and the load balancer is configured to direct traffic to the upgraded nodes as they become available for use. This allows you to upgrade FusionAuth with minimal downtime. 
+
+Because individual nodes are taken offline during the rolling update, the capacity of your installation is reduced during the upgrade.
+
+You can choose to run in-place upgrades on each node or to replace each node with a new node running the latest version of FusionAuth. In-place upgrades are potentially faster, but if something goes wrong, you may need to roll back the upgrade. Replacing nodes may be slower, but if something goes wrong, you can simply revert to the previous version of FusionAuth by pointing the load balancer to the old nodes.
+
+A rolling upgrade procedure might look like this:
+
+- Stop the first node and remove it from the load balancer.
+- Upgrade FusionAuth on the first node. Once FusionAuth is updated, start it up so that it can perform the database migrations. FusionAuth will lock the database and own it while the database upgrade is completed. Note that there will be a schema mismatch between the first node and the other nodes, so the other nodes may give errors while the first node is being upgraded. The nature of these errors depends on the code paths being updated as well as how users interact with the system. They should be minimal.
```

Comment:
```suggestion
- Upgrade FusionAuth on the first node. Once FusionAuth is updated, start it up so that it can perform the database migrations. FusionAuth will lock the database and own it while the database upgrade is completed. Note that there will be a schema mismatch between the first node and the other nodes, so the other nodes may give errors while the first node is being upgraded. The nature of these errors depends on the code paths being updated as well as how users interact with the system. For core login functionality, they should be minimal.
```